packages/lit/build.sh

```diff
@@ -0,0 +1,25 @@
+TERMUX_PKG_HOMEPAGE=https://luvit.io
+TERMUX_PKG_DESCRIPTION="Toolkit for developing, sharing, and running luvit/lua programs and libraries."
+TERMUX_PKG_LICENSE="Apache-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=3.8.5
+TERMUX_PKG_SRCURL=git+https://github.com/luvit/lit.git
+TERMUX_PKG_GIT_BRANCH="${TERMUX_PKG_VERSION}"
+TERMUX_PKG_BUILD_DEPENDS="luvi"
+TERMUX_PKG_SUGGESTS="luvi, luvit"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_NO_STRIP=true
+
+termux_step_configure() {
+	curl -L https://github.com/luvit/lit/raw/master/get-lit.sh | sh
```

Comment:
This should probably be pulled from the same tag matching `$TERMUX_PKG_VERSION`.
Also wouldn't that already be in the source repo we just cloned?

Reply:
It's needed to build `lit` and `luvit`. Think of it as `termux_setup_*` to fetch toolchain.

Reply:
But didn't we ***just*** clone `https://github.com/luvit/lit.git` and checkout the appropriate tag in the download step?

`get-lit.sh` should already be present at `$TERMUX_PKG_SRCDIR/get-lit.sh`

Reply:
I just realized it's command mistake, it's should be `-Lo-` instead.

Reply:
Fixed in commit 1e0700753.

Reply:
Again, this is `lit`'s build.
We already have `get-lit.sh` at `$TERMUX_PKG_SRCDIR/get-lit.sh`

Reply:
Can you give me the context about `get-lit.sh`? I'm very confused here.

`get-lit.sh` is just a shell script to fetch `luvi`, `lit`, `luvit`. It has nothing to do with the package besides using `lit` binary from `get-lit.sh` to build the package.

Reply:
This file is `lit/build.sh`
We're in the `termux_step_configure()` function.
Which runs after `termux_step_get_source()`.
So far so clear?

So we already have `lit`'s source repo cloned into `$TERMUX_PKG_SRCDIR`.
Why would we then go and download `get-lit.sh`, from `https://github.com/luvit/lit/raw/master/get-lit.sh`?
When we just cloned `https://github.com/luvit/lit.git`, which already contains that file.

Reply:
`get-lit.sh` != `https://github.com/luvit/lit.git`. `get-lit.sh` purpose is to fetch all pre-built binaries. In this case it;s used to fetch pre-built binaries and use it to build `lit`, as `https://github.com/luvit/lit.git` itself doesn't contain the binary to build itself.

https://github.com/luvit/lit/blob/master/get-lit.sh#L19

Reply:
How are they not the same file?
They literally come from the same place.